src/baseline.rs

```diff
@@ -244,25 +255,54 @@ impl BaselineLoader for RegistryBaseline {
         ));
         std::fs::create_dir_all(&base_root)?;
         let manifest_path = base_root.join("Cargo.toml");
-        std::fs::write(
-            &manifest_path,
-            format!(
-                "
-[workspace]
-
-[package]
-name = 'rustdoc'
-version = '0.0.0'
-publish = false
 
-[lib]
-path = 'lib.rs'
+        let manifest = {
+            use cargo_toml::*;
+            // tonowak to obi1kenobi: is there an easy way of using cargo_toml::*, but without
+            // bringing it to the scope of the whole function? With those nested let I've managed
+            // to not pollute the scope, but it seems stupid.
 
-[dependencies]
-{} = '={}'
-",
-                name, base_version
-            ),
+            let manifest = Manifest::<()> {
+                package: {
+                    let mut package = Package::new("rustdoc", "0.0.0");
+                    package.publish = Inheritable::Set(Publish::Flag(false));
+                    Some(package)
+                },
+                workspace: Some(Workspace::<()>::default()),
+                lib: {
+                    let product = Product {
+                        path: Some("lib.rs".to_string()),
+                        ..Product::default()
+                    };
+                    Some(product)
+                },
+                dependencies: {
+                    let project_with_features = DependencyDetail {
+                        version: Some(base_version.clone()),
+                        // adding features fixes:
+                        // https://github.com/obi1kenobi/cargo-semver-check/issues/147
+                        features: crate_baseline
+                            .features()
+                            .iter()
+                            .map(|(key, _values)| key.clone())
+                            .collect(),
+                        ..DependencyDetail::default()
+                    };
+                    let mut deps = DepsSet::new();
+                    deps.insert(
+                        name.to_string(),
+                        Dependency::Detailed(project_with_features),
+                    );
+                    deps
+                },
+                ..Default::default()
+            };
+            manifest
+        };
+        std::fs::write(
+            &manifest_path,
+            toml::to_string(&manifest)?.replace("edition = \"2021\"", ""),
+            // tonowak to obi1kenobi: do you know why this replace is necessary?
```

Comment:
> I'm not sure, why did you need to add it? :) Would it cause any harm to leave the edition key in the TOML?

I had an error without removing the edition, I'll try to reproduce the error and paste it here when I'll have some time.

> Also, consider making these comments as GitHub comments on the PR itself for future PRs? That way we can't miss them either in terms of discussing them or in terms of remembering to remove them later when they've been sorted out.

Sounds reasonable, but then I could forget to comment something after creating the PR...

Reply:
> Sounds reasonable, but then I could forget to comment something after creating the PR...

You could perhaps keep these notes in a personal local file, and then turn them into comments as you open the PR? That way they aren't in the source code but they are written down before opening the PR.

I usually keep a notes file like this for myself to keep track of everything that's in progress, and I'd recommend it as a process in general.

Reply:
> I had an error without removing the edition, I'll try to reproduce the error and paste it here when I'll have some time.

Error: Failed when running cargo-doc on ../rust-libp2p/target/semver-checks/registry-libp2p_core-0_37_0/Cargo.toml: error: cannot specify features for packages outside of workspace

Reply:
Is it perhaps getting confused about the fact that the `target` dir is inside the workspace of the other `Cargo.toml`? Here's what I'd suggest:
- Try reproducing the error with a manual call to cargo from the terminal.
- Then try moving the `target` directory somewhere else, outside of the workspace of `libp2p`, and try reproducing the error again with the same call.

I'm a bit worried about the edition removal because it feels like there might be a deeper issue here and the edition is just a symptom rather than the cause.

Reply:
Minimal command that breaks (ran in the folder with the generated Cargo.toml):
`cargo doc --all-features --package libp2p-core`

Reply:
Setting `--target-dir=/tmp/target` doesn't help. Do you have an idea what is wrong? I don't know how to progress from here.

Reply:
The smallest breaking Cargo.toml seems to be
```toml
[package]
name = "rustdoc"
edition = "2021"
version = "0.0.0"

[workspace]

[dependencies]
libp2p = "0.37.0"

[lib]
path = "lib.rs"
```
with an empty `lib.rs`.

Reply:
Re-reading the error message one more time, I think the issue is about attempting to combine `--all-features` with `--package libp2p-core`. The `libp2p-core` package is a dependency of the project and as such its features are specified in the `Cargo.toml`, so we can't run `--all-features` on it because what if the `Cargo.toml` entry for it specifies some other set of features? In other words, `--all-features` only makes sense in the context of commands applied to the source code corresponding to the `Cargo.toml`, and does not make sense for dependency packages where features are specified elsewhere.

I think the answer is to not include `--all-features` in the "make fake project + get rustdoc of the dependency" branch of `cargo-semver-checks`.

Reply:
I think this is the only remaining item for this PR, otherwise it's ready to merge.

Reply:
Sure, I'll do it in 2-3h

Reply:
No rush, whenever you have a moment is fine.